### DISCUSSION TITLE
Cannot access the server outside the container

### DISCUSSION THREAD

**Discussion by aminvandh** [fastapi/fastapi#11163] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
LISTENING_IP: str = EnvSettings.get_str_var(r'FASTAPI_IP', r'0.0.0.0')
LISTENING_PORT: int = EnvSettings.get_int_var(r'FASTAPI_PORT', 8000)
```

### Description

I cannot access it outside the container (docker, nor lcx) while i can easily open apache, why ?

### Operating System

Linux

### Operating System Details

Host ubuntu 
guest ubuntu lxc

### FastAPI Version

0.108.0

### Pydantic Version

1.10.13

### Python Version

3.10

**Comment by ahmedabdou14** [comment]
Can you show us the command you use to run FastApi and your Dockerfile

**Reply by aminvandh** [reply]
This code successfuly runs the api but it's not reachable from outside (i use vnc):

```
FROM debian:bookworm

LABEL version="2023.07.30"

SHELL ["/bin/bash", "-c"]

EXPOSE 6612/tcp
EXPOSE 8000/tcp

ARG DATABASE=sqlite

ENV DEBUG="0"
ENV FASTAPI_IP="0.0.0.0"
ENV FASTAPI_PORT="8000"

COPY . /API-Server/
WORKDIR /API-Server/

# SETUP DEPENDENCIES
ENV DEBIAN_FRONTEND="noninteractive"
RUN apt -qq -y update && apt install -qq --yes --no-install-recommends apt-utils libncurses5-dev perl-modules-5.36 && apt -qq -y upgrade && apt install -qq --yes --no-install-recommends git gcc binutils python3 libpython3-dev python3-pip python3-venv make sqlite3 libmariadb3 libmariadb-dev libsqlite3-dev mariadb-server sendmail mariadb-client locales sudo unzip zip curl wget
RUN make venv
RUN apt clean && rm -f -r /var/lib/apt/lists/*

# SETUP DB CONFIG
RUN echo "[client]\
default-character-set=utf8mb4\
[mysqld]\
#innodb_force_recovery=4\
# BASIC SETTINGS # \
bind-address=0.0.0.0\
datadir=/var/lib/mysql\
pid-file=/var/run/mysqld/mysqld.pid\
port=6612\
socket=/var/run/mysqld/mysqld.sock\
user=mysql\
character-set-server=utf8mb4\
collation-server=utf8mb4_unicode_ci\
default-time-zone=+00:00\
default_storage_engine=InnoDB\
default_week_format=0\
large_pages=1\
local_infile=1\
tmpdir=/tmp\
# LOGGING & REPLICATION #\
binlog_cache_size=128M\
binlog_row_metadata=NO_LOG\
binlog_stmt_cache_size=128M\
#expire_logs_days=1\
#log_bin=1\
log_error=/var/log/mysql/error.log\
log_output=FILE\
log_queries_not_using_indexes=1\
log_warnings=1\
long_query_time=4\
max_binlog_cache_size=32M\
max_binlog_size=10M\
max_binlog_stmt_cache_size=32M\
slow_query_log=0\
slow_query_log_file=/var/log/mysql/slow_queries.log" > /etc/mysql/mariadb.cnf

RUN sed -i 's|3306|6612|g' /etc/mysql/mariadb.conf.d/50-server.cnf

# SETUP DATABASE
RUN export PATH="./venv/bin:${PATH}"
RUN service mariadb start && mysql -u root -e "SET SQL_MODE = 'NO_AUTO_VALUE_ON_ZERO';START TRANSACTION;SET time_zone = '+00:00';CREATE DATABASE IF NOT EXISTS \`API_Server\` DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;USE \`API_Server\`;COMMIT;\
CREATE USER 'api-server'@'%' IDENTIFIED BY 'P@sW0rd_2023!'; \
GRANT SELECT, INSERT, UPDATE, DELETE, CREATE, DROP, RELOAD, FILE, INDEX, ALTER, SUPER, CREATE TEMPORARY TABLES, EXECUTE, CREATE VIEW, SHOW VIEW, CREATE ROUTINE, ALTER ROUTINE, EVENT, TRIGGER ON *.* TO 'api-server'@'%' IDENTIFIED BY 'P@sW0rd_2023!';\
FLUSH PRIVILEGES;"
RUN if [ "$DATABASE" = 'mariadb' ]; then make mariadb_database; else make sqlite3_database; fi

# RUN THE SERVER
ENTRYPOINT make run_server
```

**Reply by n8sty** [reply]
Are you publishing the correct container port? `docker run -p 8000:8000 yourcontainer`

**Reply by GodMoonGoodman** [reply]
When containerizing from a Docker image, port mapping is required to access from outside. @n8sty gave good example.

**Reply by aminvandh** [reply]
Yes and didn't work. I think there might be something with my code.

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean and organized.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
